### PR TITLE
Restore getPassword compatibility

### PR DESCRIPTION
**Pull request by timja**
Tested with irc-bot plugin and all seems fine,
cc @jimklimov

### CODE REVIEW

File: src/main/java/hudson/plugins/im/GroupChatIMMessageTarget.java [hunk 1]
```diff
@@ -57,15 +65,24 @@ public String getName() {
...
     }
 
     public Secret getSecretPassword() {
-        return secretPassword;
+        return password;
+    }
+
+    @DataBoundSetter
+    public void setSecretPassword(Secret secretPassword) {
+        if (Secret.toString(secretPassword).equals("")) {
```
Comment (timja) [inline src/main/java/hudson/plugins/im/GroupChatIMMessageTarget.java:77]: stapler submits empty string which makes it look like this has a password, meaning it shows up in jcasc and in the configure ui as with a password.

with this fix that doesn't happen

```
  ircPublisher:
    charset: "UTF-8"
    commandPrefix: "!jenkins"
    defaultTargets:
    - groupChat:
        name: "#hello"
        notificationOnly: false
    - groupChat:
        name: "#private"
        notificationOnly: false
        secretPassword: "{AQAAABAAAAAQXRHSKt9oLho8NeWn1Yt1qhlsYvvrZ4b58vJ3W4l69tk=}"
```

**Review by daniel-beck — APPROVED** [review]
Looks reasonable. Untested, should be tested with both 1.39 and earlier, and 1.40/1.41 storage to be safe.

**Comment by timja** [comment]
> Looks reasonable. Untested, should be tested with both 1.39 and earlier, and 1.40/1.41 storage to be safe.

I've tested with 1.40 storage, g2g now but feel free to test with older storage someone

**Review by jimklimov — APPROVED** [review]
LGTM too but can't quickly test vs older plugin  - my jenkins is updated,  will try to make a scratch one. :)

**Comment by timja** [comment]
@jimklimov :shipit: ?

**Comment by jimklimov** [comment]
ouch, sorry, got consumed by the git refrepo fanout quest

**Pull request merged**